connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/internal/writer/BigtableWriterTest.java

```diff
@@ -0,0 +1,178 @@
+    @After
+    public void close() {
+        client.close();
+        tableAdminClient.close();
+    }
+
+    /**
+     * Tests the writing of data to a Bigtable table using the {@link BigtableSinkWriter}.
+     *
+     * @throws IOException If an error occurs while writing data.
+     * @throws InterruptedException If the thread is interrupted while waiting for write completion.
+     * @throws ExecutionException If an error occurs during the write operation.
+     */
+    @Test
+    public void testWriter() throws IOException, InterruptedException, ExecutionException {
+        Schema schema =
+                SchemaBuilder.builder()
+                        .record("WriterTest")
+                        .fields()
+                        .requiredString(TestingUtils.ROW_KEY_FIELD)
+                        .requiredString(TestingUtils.STRING_FIELD)
+                        .requiredInt(TestingUtils.INTEGER_FIELD)
+                        .endRecord();
+
+        GenericRecordToRowMutationSerializer serializer =
+                GenericRecordToRowMutationSerializer.builder()
+                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                        .withColumnFamily(TestingUtils.COLUMN_FAMILY)
+                        .build();
+
+        BigtableSinkWriter<GenericRecord> writer =
+                new BigtableSinkWriter<GenericRecord>(
+                        new BigtableFlushableWriter(client, null, TestingUtils.TABLE), serializer);
+
+        for (int i = 0; i < MAX_RANGE; i++) {
+            GenericRecord testRecord = new GenericData.Record(schema);
+            testRecord.put(TestingUtils.ROW_KEY_FIELD, "key" + i);
+            testRecord.put(TestingUtils.STRING_FIELD, "string" + i);
+            testRecord.put(TestingUtils.INTEGER_FIELD, i);
+            writer.write(testRecord, null);
+        }
+        checkEmptyData();
+        writer.flush(false);
+        checkWrittenData();
+        writer.close();
+    }
+
+    /**
+     * Helper method to verify that the Bigtable table is empty before writing data.
+     *
+     * @throws InterruptedException If the thread is interrupted while waiting for the read
+     *     operation.
+     * @throws ExecutionException If an error occurs during the read operation.
+     */
+    private void checkEmptyData() {
+        for (int i = 0; i < MAX_RANGE; i++) {
+            Row row = client.readRow(TableId.of(TestingUtils.TABLE), "key" + i);
+            assertNull(row);
+        }
+    }
+
+    /**
+     * Helper method to verify that the data has been correctly written to the Bigtable table.
+     *
+     * @throws IOException If an error occurs while reading data from the table.
+     * @throws InterruptedException If the thread is interrupted while waiting for the read
+     *     operation.
+     * @throws ExecutionException If an error occurs during the read operation.
+     */
+    private void checkWrittenData() {
```

Comment:
same as my other comment, we really want to avoid doing assertions in helpers. makes the test a lot harder to interpret

Reply:
Given the assertion is per read row, I would have to move this to the main test. This is not a problem from me, done